### PR TITLE
Fix negative unary operation for boolean type

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Negative unary op on boolean series is resulting in conversion to `int` type:
```python
In [1]: import cudf

In [2]: s = cudf.Series([True, False])

In [3]: s
Out[3]: 
0     True
1    False
dtype: bool

In [4]: -s
Out[4]: 
0   -1
1    0
dtype: int8

In [5]: -s.to_pandas()
Out[5]: 
0    False
1     True
dtype: bool
```
The PR fixes the above issue by returning inversion of the boolean column instead of multiplying with `-1`.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_unaops.py [hunk 1]
```diff
@@ -123,3 +123,9 @@ def test_scalar_no_negative_bools():
         ),
     ):
         -x
+
+
+def test_series_bool_neg():
+    sr = Series([True, False, True, False, True, True])
```
Comment (mroeschke) [inline python/cudf/cudf/tests/test_unaops.py:129]: Maybe good to test with NA too?

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -1891,8 +1891,17 @@ def __rmatmul__(self, other):
         return self.dot(other, reflect=True)
 
     # Unary logical operators
+    @_cudf_nvtx_annotate
```
Comment (bdice) [inline python/cudf/cudf/core/frame.py:1894]: Can we annotate everything in this file, or drop the annotation here? Just want consistency. Fine to do that in a separate PR.

File: python/cudf/cudf/tests/test_unaops.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_unaops.py]: Done 👍

File: python/cudf/cudf/core/frame.py (hunk 2 above)
Comment (galipremsagar) [reply python/cudf/cudf/core/frame.py:1914]: Annotated in this PR 👍

**Review by bdice — APPROVED** [review]

**Review by wence- — APPROVED** [review]
